runtime/jvmti/jvmtiStackFrame.c

```diff
@@ -283,16 +283,25 @@ jvmtiGetFrameCount(jvmtiEnv* env,
 		rc = getVMThread(currentThread, thread, &targetThread, TRUE, TRUE);
 		if (rc == JVMTI_ERROR_NONE) {
 			J9StackWalkState walkState;
-
-			vm->internalVMFunctions->haltThreadForInspection(currentThread, targetThread);
-
-			walkState.walkThread = targetThread;
 			walkState.flags = J9_STACKWALK_INCLUDE_NATIVES | J9_STACKWALK_VISIBLE_ONLY;
 			walkState.skipCount = 0;
-			vm->walkStackFrames(currentThread, &walkState);
+
+#if JAVA_SPEC_VERSION >= 19
+			if ((NULL == targetThread) && IS_VIRTUAL_THREAD(currentThread, J9_JNI_UNWRAP_REFERENCE(thread))) {
```

Comment:
`J9_JNI_UNWRAP_REFERENCE(thread)` is used both in this line and the following line. You can move `j9object_t threadObject` outside of this `if {` scope and so that you do `J9_JNI_UNWRAP_REFERENCE(thread)` only once.

Reply:
I guess `J9_JNI_UNWRAP_REFERENCE(thread)` will segfault if thread is NULL.

Reply:
> I guess `J9_JNI_UNWRAP_REFERENCE(thread)` will segfault if thread is NULL.

If thread is NULL the targetThread will be currentThread, so `J9_JNI_UNWRAP_REFERENCE` won't run and that might be a reason that we don't want to move `J9_JNI_UNWRAP_REFERENCE` out of the if statement?

Reply:
OK, please put a comment in the code explaining this.

Reply:
you can also split the if statement to
```c++
if (NULL == targetThread) {
    j9object_t threadObject = J9_JNI_UNWRAP_REFERENCE(thread);
    if (IS_VIRTUAL_THREAD(currentThread, threadObject) {
        ...
    }
}
```